src/jit/morph.cpp

```diff
@@ -3439,23 +3438,12 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
             //
             // Figure out if the argument will be passed in a register.
             //
-            bool passedInRegisters = true;
-#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
-            passedInRegisters = !isStructArg;
-            if (!passedInRegisters)
-            {
-                if (structDesc.passedInRegisters)
-                {
-                    passedInRegisters = true;
-                }
-                else
-                {
-                    passedInRegisters = false;
-                }
-            }
 
+            if (isRegParamType(genActualType(argx->TypeGet()))
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+                && (!isStructArg || structDesc.passedInRegisters)
 #endif
-            if (passedInRegisters && isRegParamType(genActualType(argx->TypeGet())))
+                )
             {
 #ifdef _TARGET_ARM_
```

Comment:
Maybe a follow-up cleanup. The issue with the `_TARGET_ARM64_` one is that it was checking zero/non-zero, not defined(), which should have been a bug (but apparently the compilers accepted it, even if not defined?)